### PR TITLE
Experiment/component isolation mode

### PR DESCRIPTION
**Pull request by Rheeseyb**
Experiment for #686 

Double clicking an instance of a component **on the canvas** will switch to an "isolation" mode, which allows you to directly edit that component. Very rough around the edges, but this works by slipping a new scene into the transient state of the canvas.

Basic sample project: https://utopia.pizza/p/2d8b43ad-pitch-ankle/?branch_name=experiment/component-isolation-mode

### CODE REVIEW

File: editor/src/core/model/project-file-utils.ts [hunk 1]
```diff
@@ -178,6 +180,47 @@ export function getOrDefaultScenes(parsedSuccess: ParseSuccess): UtopiaJSXCompon
...
+): Array<TopLevelElement> {
+  // Run through the old top level elements, replacing the exported elements with those in the
+  // newly updated result with the same name.
+  // If it doesn't exist in the updated result, delete it.
+  // For any new items in the updated result, add them in.
+  const addedSoFar: Set<string> = Utils.emptySet()
+  let newTopLevelElements: Array<TopLevelElement> = []
+  Utils.fastForEach(topLevelElements, (oldTopLevelElement) => {
+    if (isUtopiaJSXComponent(oldTopLevelElement)) {
+      const updatedElement = newUtopiaComponents.find((e) => e.name === oldTopLevelElement.name)
```
Comment (seanparsons) [inline editor/src/core/model/project-file-utils.ts:205]: Super minor, but might help a little with clarity.
```suggestion
      const updatedComponent = newUtopiaComponents.find((e) => e.name === oldTopLevelElement.name)
```

**Review by seanparsons — APPROVED** [review]

**Comment by enidemi** [comment]
I really how smooth this is! It could be improved with adding more visual indicators to make it clear that I'm editing the original component.

**Comment by alecmolloy** [comment]
Its been a long time since i've tooted the horn of the isolation mode, but I am glad to see this coming up from someone else other than me. :) Curious to discuss this one more in detail, want to understand "what" the user is editing and how to best communicate that.

**Comment by enidemi** [comment]
Merged to https://github.com/concrete-utopia/utopia/pull/727

**Pull request closed without merging**